src/storage/txn/actions/flashback_to_version.rs

```diff
@@ -62,21 +62,29 @@ pub fn flashback_to_version_read_write<S: Snapshot>(
     let mut key_old_writes = Vec::with_capacity(FLASHBACK_BATCH_SIZE - key_locks_len);
     // Check the latest commit ts to make sure there is no commit change during the
     // flashback, otherwise, we need to abort the flashback.
-    for (key, commit_ts, old_write) in key_ts_old_writes {
-        if commit_ts > flashback_commit_ts {
+    for (key, commit_ts, old_write) in key_ts_old_writes.iter() {
+        if *commit_ts > flashback_commit_ts {
             return Err(Error::from(ErrorInner::InvalidTxnTso {
                 start_ts: flashback_start_ts,
                 commit_ts: flashback_commit_ts,
             }));
         }
-        // Since the first flashback preparation phase make sure there will be no writes
-        // other than flashback after it, so we need to check if there is already a
-        // successful flashback result, and if so, just finish the flashback ASAP.
-        if commit_ts == flashback_commit_ts {
-            key_old_writes.clear();
-            return Ok((key_old_writes, false));
+        // Although the first flashback preparation phase make sure there will be no
+        // writes other than flashback after it, we can't return it ASAP.
+        // Suppose the second phase procedure contains two batches to flashback. After
+        // the first batch is committed, if the region is down, client will retry the
+        // flashback from the very first beginning, because the data in the first batch
+        // has been written the flashbacked data with the same commit_ts, So we need to
+        // skip it to make sure to keep writing the data that is needed.
```

Comment:
```suggestion
        // Although the first flashback preparation phase makes sure there will be no
        // writes other than flashback after it, we CAN NOT return directly here.
        // Suppose the second phase procedure contains two batches to flashback. After
        // the first batch is committed, if the region is down, the client will retry the
        // flashback from the very first beginning, because the data in the first batch
        // has been written the flashbacked data with the same `commit_ts`, So we need to
        // skip it to ensure the following data will be flashbacked continuously.
```